sdk/samples/iot/paho_iot_pnp_with_provisioning_sample.c

```diff
@@ -250,13 +250,15 @@ static void register_device_with_provisioning_service(void)
   uint8_t mqtt_payload[MQTT_PAYLOAD_BUFFER_LENGTH];
   size_t mqtt_payload_length;
 
+  AZ_PUSH_IGNORE_DEPRECATIONS
   rc = az_iot_provisioning_client_get_request_payload(
       &provisioning_client,
       custom_registration_payload_property,
       NULL,
       mqtt_payload,
       sizeof(mqtt_payload),
       &mqtt_payload_length);
+  AZ_POP_WARNINGS
```

Comment:
are we able to use the new API instead?

Reply:
Yes, we could if we needed to in the future. For now, we'll leave these as-is for reference (i.e., existing applications picking up the SDK with the deprecation).